app/settings/surveys/survey-editor.directive.js

```diff
@@ -472,10 +472,61 @@ function SurveyEditorController(
             ModalService.close();
         }
         $scope.editAttribute = attribute;
-        var title = attribute.id ? 'survey.edit_field' : 'survey.add_field';
+        var fieldType = getFieldType(attribute) ? getFieldType(attribute) : attribute.label;
+        var title = attribute.id ? $translate.instant('survey.edit_field', {fieldType}) : $translate.instant('survey.add_field', {fieldType});
+        title = title.replace('&amp;', '&');
         ModalService.openTemplate('<survey-attribute-editor></survey-attribute-editor>', title, '', $scope, true, true);
     }
 
+    function getFieldType(attribute) {
+        var fieldTypes = {
+            'text': {
+                'varchar': $translate.instant('survey.short_text')
+            },
+            'textarea': {
+                'text': $translate.instant('survey.long_text')
+            },
+            'number': {
+                'decimal': $translate.instant('survey.number_decimal'),
+                'int': $translate.instant('survey.number_integer')
+            },
+            'location': {
+                'point': $translate.instant('survey.location')
+            },
+            'date': {
+                'datetime': $translate.instant('survey.date')
+            },
+            'datetime': {
+                'datetime': $translate.instant('survey.datetime')
+            },
+            'select': {
+                'varchar': $translate.instant('survey.select')
+            },
+            'radio': {
+                'varchar': $translate.instant('survey.radio_button')
+            },
+            'checkbox': {
+                'varchar': $translate.instant('survey.checkbox')
+            },
+            'relation' : {
+                'relation': $translate.instant('survey.related_post')
+            },
+            'upload' : {
+                'media': $translate.instant('survey.upload_image')
+            },
+            'video' : {
+                'varchar': $translate.instant('survey.embed_video')
+            },
+            'markdown' : {
+                'markdown': 'Markdown'
+            },
+            'tags' : {
+                'tags': $translate.instant('survey.categories')
+            }
+        }
+        return fieldTypes[attribute.input][attribute.type];
```

Comment:
I'd probably suggest checking fieldTypes[attribute.input] exists just in case to avoid an error throwing.